packages/block-editor/src/components/inserter/hooks/use-insertion-point.js

```diff
@@ -92,10 +93,20 @@ function useInsertionPoint( {
 				// Insert after a specific client ID.
 				_destinationIndex = getBlockIndex( clientId );
 			} else if ( ! isAppender && selectedBlockClientId ) {
-				_destinationRootClientId = getBlockRootClientId(
-					selectedBlockClientId
-				);
-				_destinationIndex = getBlockIndex( selectedBlockClientId ) + 1;
+				const sectionRootClientId = getSectionRootClientId();
+
+				if ( sectionRootClientId === selectedBlockClientId ) {
+					_destinationRootClientId = sectionRootClientId;
+					_destinationIndex = getBlockOrder(
+						_destinationRootClientId
+					).length;
```

Comment:
So as you described, the inserter is empty is because it usually inserts after the selected (post content) block, but the block list that the post content block is in is locked.

The code here detects that case for the section root client id (which is again the post content block in this case) and special cases it to insert within that block instead of after it.

Are there any other cases where this might be a problem? In some cases, `sectionRootClientId` can be the block that's the `<main>` element, so I think this logic might also apply there.

I wonder if it's something that can be made more generic - that if there's a container block selected and we can't insert after it (the root is templateLocked or blockEditingMode is non-default), then instead we try inserting inside it. This would be quite a fundamental change in the middle of the 6.7 cycle, but it might also be safer and have less edge cases.

The alternative could be that the special case is kept, but it's only active during zoom out mode, which I don't think would cause any issues.

I think separately the empty state of the pattern inserter could be improved, because there are probably cases where it's legitimately impossible to insert blocks and it might be empty (e.g. if entire posts or pages are locked).

Reply:
> The alternative could be that the special case is kept, but it's only active during zoom out mode, which I don't think would cause any issues.

I think this might be the best approach. 

I agree that changing the way Pattern insertion works at this stage in the release is too large a change to feel comfortable.

Reply:
> Are there any other cases where this might be a problem? In some cases, sectionRootClientId can be the block that's the <main> element, so I think this logic might also apply there.

Yeah, I can repro this, it does insert inside the `<main>`, but most of the time I think it's where the user would want to insert, so maybe it's better. It just makes the insertion behavior a little inconsistent!

Reply:
I updated to target Zoom Out only.